src/components/CmsZone.vue

```diff
@@ -93,7 +96,51 @@ import CmsContent from './CmsContent';
 import CmsInspectSheet from './CmsInspectSheet.vue';
 
 const durationVisibleToBeTrackedMs = 1000;
-const percentVisible = 50;
+const percentVisible = 0.5;
+
+const zones: { [key: string]: () => Promise<void> } = {};
+
+const trackers: { [key: string]: () => void } = {};
+
+let fetchObserver: IntersectionObserver;
+let trackObserver: IntersectionObserver;
+let checkObserver: IntersectionObserver;
+let observersCreated = false;
+
+function createObservers() {
+  if (observersCreated) {
+    return;
+  }
+
+  observersCreated = true;
+  fetchObserver = new IntersectionObserver((entries, observer) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting && zones[entry.target.id]) {
+        observer.unobserve(entry.target);
+        zones[entry.target.id]();
+      }
+    });
+  }, { rootMargin: '25%' });
+
+  trackObserver = new IntersectionObserver((entries) => {
+    entries.forEach((entry) => {
+      if (entry.intersectionRatio >= percentVisible && trackers[entry.target.id]) {
+        setTimeout(() => checkObserver.observe(entry.target), durationVisibleToBeTrackedMs);
+      }
+    });
+  }, { threshold: percentVisible });
+
+  checkObserver = new IntersectionObserver((entries, observer) => {
```

Comment:
why do we need another observer for the check? can we do like 
```ts
  trackObserver = new IntersectionObserver((entries, observer) => {
    entries.forEach((entry) => {
      if (entry.intersectionRatio >= percentVisible && trackers[entry.target.id]) {
        setTimeout(() => {
             if (entry.intersectionRatio >= percentVisible && trackers[entry.target.id]) {
               observer.unobserve(entry.target);
               trackers[entry.target.id]();
               delete trackers[entry.target.id];
             }
        }, durationVisibletoBeTrackedMs);
      }
    });
  }, { threshold: percentVisible });
```

Reply:
The entry has a timestamp and isn't updated in real time.